src/components/site/educatorModeToggle/index.jsx

```diff
@@ -0,0 +1,73 @@
+import PropTypes from 'prop-types';
+import React from 'reactn';
+import classnames from 'classnames';
+import { SelectionControl, TextField } from 'react-md';
+
+const EDUCATOR_PASSPHRASE = '3ducatorMod3';
+
+class EducatorModeToggle extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      passphraseInput: null,
+      passphraseError: false,
+    };
+  }
+
+  handleEducatorModeChange = value => {
+    const { passphraseInput } = this.state;
+
+    if (value && passphraseInput === EDUCATOR_PASSPHRASE) {
+      this.setState({ passphraseError: false });
+      this.dispatch.setEducatorMode(true);
+    }
+
+    if (value && passphraseInput !== EDUCATOR_PASSPHRASE) {
+      this.setState({ passphraseError: true });
+      this.dispatch.setEducatorMode(false);
```

Comment:
should these global store dispatch events be in the callback of the component setState or does it not matter?